internal/dataapi/listener.go

```diff
@@ -69,17 +71,21 @@ func (lis *Listener) Run(ctx context.Context) {
 	}
 
 	srv := &http.Server{
-		Handler: srvHandler,
-		BaseContext: func(_ net.Listener) context.Context {
-			return ctx
+		Handler:  srvHandler,
+		ErrorLog: slog.NewLogLogger(lis.opts.L.Handler(), slog.LevelError),
+		BaseContext: func(net.Listener) context.Context {
+			return conninfo.Ctx(ctx, conninfo.New())
```

Comment:
That's the main change – we should always return context with `connInfo`

Reply:
I see, so context passed to our handler now always contains connInfo :+1: